addons/udes_stock/models/stock_quant_package.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, fields, models, _
+from collections import defaultdict
+
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class StockQuantPackage(models.Model):
+
+    _name = 'stock.quant.package'
+
+    _inherit = ['stock.quant.package', 'abstract.stock.model']
+
+    CREATE = True
+
+    # This works - but there must be a nicer way
+    name = fields.Char(
+        'Package Reference',
+        copy=False,
+        index=True,
+        default=lambda self: self.env['ir.sequence']
+        .next_by_code('stock.quant.package')
+        .replace('PACK', 'UDES')
+        or _('Unknown Pack'),
+    )
+
+    def _get_product_quantities_by_key(self, get_key=lambda q: q.product_id):
```

Comment:
This function is almost the same as `stock.quant.group_quantity_by_product()`. Shall we move stock.quant commit before this one and use it ?

Reply:
You mean just delete this function completely, and rely on stock.quant.group_quantity_by_product() if we want this behaviour?  Or just use stock.quant.group_quantity_by_product() within it?

Reply:
The latter i guess

Reply:
Do we want to pass only_available to it in that case?